minoc/src/main/java/org/opencadc/minoc/HeadAction.java

```diff
@@ -0,0 +1,130 @@
+*  OpenCADC project.                    OpenCADC.
+*
+*  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.minoc;
+
+import java.net.URI;
+
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.minoc.ArtifactUtil.HttpMethod;
+
+/**
+ * Interface with storage and inventory to get the metadata of an artifact.
+ *
+ * @author majorb
+ */
+public class HeadAction extends ArtifactAction {
+    
+    private static final Logger log = Logger.getLogger(HeadAction.class);
+
+    /**
+     * Default, no-arg constructor.
+     */
+    public HeadAction() {
+        super(HttpMethod.HEAD);
+    }
+    
+    /**
+     * Constructor for subclass, GetAction
+     * @param method The http method
+     */
+    public HeadAction(HttpMethod method) {
+        super(method);
+    }
+
+    /**
+     * Download the artifact or cutouts of the artifact.
+     * @param artifactURI The identifier for the artifact. 
+     * @return The artifact
+     */
+    @Override
+    public Artifact execute(URI artifactURI) throws Exception {
+        ArtifactDAO dao = getArtifactDAO();
+        Artifact artifact = getArtifact(artifactURI, dao);
+
```

Comment:
Making all these temp variables is pointless and makes writing clear code below harder. Just use the get methods or fields directly. For those that can be accessed via a field, setHeader should only be called when the value is not null. That's going to be more clear without local variables.